code/modules/recycling/disposal/construction.dm

```diff
@@ -0,0 +1,231 @@
+// Disposal pipe construction
+// This is the pipe that you drag around, not the attached ones.
+
+/obj/structure/disposalconstruct
+	name = "disposal pipe segment"
+	desc = "A huge pipe segment used for constructing disposal systems."
+	icon = 'icons/obj/atmospherics/pipes/disposal.dmi'
+	icon_state = "conpipe"
+	anchored = FALSE
+	density = FALSE
+	pressure_resistance = 5*ONE_ATMOSPHERE
+	level = 2
+	max_integrity = 200
+	var/obj/pipe_type = /obj/structure/disposalpipe/segment
+	var/pipename
+
+
+/obj/structure/disposalconstruct/examine(mob/user)
+	..()
+	to_chat(user, "<span class='notice'>Alt-click to rotate it clockwise.</span>")
+
+/obj/structure/disposalconstruct/New(loc, _pipe_type, _dir = SOUTH, flip = FALSE, obj/make_from)
+	..()
+	if(make_from)
+		pipe_type = make_from.type
+		setDir(make_from.dir)
+		anchored = TRUE
+
+		if(istype(make_from, /obj/structure/disposalpipe))
+			var/obj/structure/disposalpipe/D = make_from
+			if(D.construct_type)
+				pipe_type = D.construct_type
+
+	else
+		if(_pipe_type)
+			pipe_type = _pipe_type
+		setDir(_dir)
+
+	pipename = initial(pipe_type.name)
+
+	if(flip)
+		flip()
+
+	update_icon()
+
+/obj/structure/disposalconstruct/Move()
+	var/old_dir = dir
+	..()
+	setDir(old_dir) //pipes changing direction when moved is just annoying and buggy
```

Comment:
Use Moved() if possible

Reply:
**Not possible**, by the point `Moved()` is called the movement is already completed and dir is already changed. Ans we have to save the old dir here so we can restore it later.